src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -223,6 +234,15 @@ async ValueTask<int> ReadAheadWithZeroByteReadAsync()
             }
         }
 
+        private bool CheckKeepAliveTimeoutExceeded()
+        {
+            // Avoid reusing a connection if there is less than one second remaining before the timeout.
+            const int TimeoutOffsetSeconds = 1;
+
+            return _keepAliveTimeoutSeconds != 0 &&
```

Comment:
Seems like this is going to be invoked even for non-HTTP/1.0 connections, and this implementation is relying on _keepAliveTimeoutSeconds always being 0 in that case?  If I've understood correctly, at a minimum this would benefit from a comment.

Reply:
Yes, that's correct. I'll add a comment.